File: tests/fixtures/openai-realtime/openai-realtime-migrate-beta-to-ga-fixed/01-ga-no-beta-header.ts
import WebSocket from 'ws';

function setupCallerSocket(apiKey: string): WebSocket {
  const url = 'wss://api.openai.com/v1/realtime';
  const callerSocket = new WebSocket(url, {
    headers: {
      Authorization: `Bearer ${apiKey}`,
    },
  });
  return callerSocket;
}

File: tests/fixtures/openai-realtime/openai-realtime-migrate-beta-to-ga-fixed/01_ga_no_beta_header.py
import os

import websockets

WS_URL = "wss://api.openai.com/v1/realtime?model=gpt-realtime"


async def connect_caller_socket():
    async with websockets.connect(
        WS_URL,
        additional_headers={
            "Authorization": f"Bearer {os.environ.get('OPENAI_API_KEY')}",
        },
    ) as ws:
        async for message in ws:
            print(message.get("type"))
